apps/system/js/sync/sync_iac_api.js

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * This is just a temporary IAC based API to allow the creation of a Syncto
+ * client proof of concept.
+ * We will expose the ability to obtain the Sync keys (kA and kB) and the
+ * generation of the X-Client-State header. In the end, all this will only be
+ * consumed by the System app, so there will be no need to expose anything.
+ */
+
+(function() {
+
+  function sendPortMessage(portName, message) {
+    var port = IACHandler.getPort(portName);
+    if (port) {
+      port.postMessage(message);
+    }
+  }
+
+  function syncHistroyRequest(id, method, args){
```

Comment:
I would not add this in the System app. The only change needed in the System app afaics is that its current Places code needs to be exposed through IAC.

Actually, wait, not even that, right? Wasn't the latest discussion that Places code should be shared, and its `writeInProgress` variable should be removed, and replaced by using the DataStore correctly?

Then we don't need IAC at all in this PR.